packages/protocol/src/Chief.sol

```diff
@@ -78,14 +152,56 @@ contract Chief is CoreRoles, AccessControl, IChief {
     timelock = newTimelock;
     // grant admin role to new timelock address
     _grantRole(DEFAULT_ADMIN_ROLE, timelock);
-    emit TimelockUpdated(newTimelock);
+    emit UpdateTimelock(newTimelock);
   }
 
-  function setOpenVaultFactory(bool state) external onlyTimelock {
-    openVaultFactory = state;
-    emit OpenVaultFactory(state);
+  /**
+   * @notice Clears and sets the `vaults` recorded in this {Chief-_vaults}.
+   *
+   * @param vaults addresses that should be recorded
+   *
+   * @dev This method should only be used in extraordinary cases.
+   * Requirements:
+   *  - Must be called from a timelock.
+   */
+  function setVaults(address[] memory vaults) external onlyTimelock {
+    address[] memory previous = _vaults;
+    delete _vaults;
+    _vaults = vaults;
+    emit SetVaults(previous, _vaults);
```

Comment:
btw @DaigaroCota did you add this function recently and what's the rationale behind it?

Reply:
I am having problems with the test for factory deployment. In particular, it fails when trying to deploy a BorrowingVault, but I think it is a problem with Foundry, since the same code was working before. Anyhow, to test vault pausability from the Chief, all vaults must be deployed using the `deployVault()` method and I realized this was a problem. There is no other way to record vaults in the {Chief} other than using this method. Considering a case in where we have to "clean" obsolete vaults (for example we uprades or so), we will require such a function.

Reply:
For that reason, I refactored the vault tests so that we are isolating the functionalities we are testing. The pausability tests and the rebalancing tests were using the deploy factories and were failing because of this Foundry problem (perhaps is something else but I still need to figure it out).

Reply:
Update. I found my error. I commented a line in LibSSTORE2 that I should not had. Deployments from factory now work, and we have cleaner refactored vault tests. 👯

Reply:
I was assumed it was not needed because of this statement:
https://github.com/Fujicracy/fuji-v2/pull/267#issuecomment-1399598921